src/main/java/org/assertj/core/api/junit/jupiter/InjectSoftAssertions.java

```diff
@@ -0,0 +1,12 @@
+package org.assertj.core.api.junit.jupiter;
+
+import static java.lang.annotation.ElementType.FIELD;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+@Retention(RUNTIME)
+@Target(FIELD)
+public @interface InjectSoftAssertions {
```

Comment:
At the moment it's any concrete instance of `SoftAssertionsProvider` with a default constructor, *or* a `ThreadLocal<T>` where `T` is any concrete instance of `SoftAssertionsProvider` with a default constructor. But I'm expecting that will change if I can hide the `ThreadLocal` functionality.

Reply:
Btw, I always intended to add code examples, but I wanted to get through the review stage before I invested too much time in examples. Our usage patterns might change through the review process and I wanted to avoid having to rewrite the examples in that event.